### ISSUE TITLE
Confusion between composer and artist.

### ISSUE THREAD

**Issue by daverz** [darrell-k/slimserver#11]
![Screenshot from 2024-03-16 22-51-42](https://github.com/darrell-k/slimserver/assets/2717422/1109900e-f955-47ab-bc53-e9b823b14c30)

The composer is (Paul) Moravec.  The artist is pianist (Ivan) Moravec.  Moravec the composer has no works on the discs with the Beethoven, Brahms, Janacek etc.   Let me know if what debug info you might need.

**Comment by darrell-k** [comment]
Fix pushed. It will now honour role types set in Additional Browse Modes for those menus. The general Artist menus will still show all works the searched-for artist is associated with.

Please confirm this fixes the issue.

**Comment by daverz** [comment]
That fixed the Classical Works, thanks, but there's still a bogus entry under "Appearances".

![Screenshot from 2024-03-17 13-09-03](https://github.com/darrell-k/slimserver/assets/2717422/2e09fcbc-f402-473b-a34b-5b8500349515)

**Comment by darrell-k** [comment]
The appearances category is specifically designed to show albums on which the artist has the role TRACKARTIST. The same would apply to the CONDUCTOR role, for example. 

My opinion is that this doesn't need changing, because if the user clicks on "Appearances", "Conductor", or whatever, they'd expect to see albums where the artist has that role, regardless of the fact that their browsing started from a role-specific menu like Composers.

I'm open to discussion, but changes in this area would require a wider discussion, as it is current released functionality. Perhaps @michaelherger would care to comment?

**Comment by daverz** [comment]
Yes, I think it's a systemic LMS issue.  In everyday language, a composer is an "artist".   Discographically, an artist is a performer, but a composer in his role as composer is not a performer.  This issue can be closed, thanks.

**Comment by darrell-k** [comment]
Actually, there is some control available for this using the "Filter Contributor Roles at the Album & Track Level" setting. Have a play with it.

But this whole area is confusing, in my opinion. There are too many inter-dependencies between different settings, or at least they're not intuitive to use, quite apart from your very valid point about the difference between the composer role and all others.

It needs more thought, so I'm going to leave this issue open for now.

**Comment by michaelherger** [comment]
Yes, there is that "old" option you mention, which I'm not sure is fully respected in the releases code...
